### PR TITLE
control/controlclient: send load balancing hint HTTP request header

### PR DESCRIPTION
**Pull request by bradfitz**
Updates tailscale/corp#1297

### CODE REVIEW

File: tailcfg/tailcfg.go [hunk 1]
```diff
@@ -2683,3 +2683,15 @@ type EarlyNoise struct {
 	// the client to prove possession of a wireguard private key.
 	NodeKeyChallenge key.ChallengePublic `json:"nodeKeyChallenge"`
 }
+
+// LBHeader is the HTTP request header used to provide a load balancer or
+// internal reverse proxy with information about the request body without
+// information about the request body with them needing to read to body. Think
```
[the remark this answers is not in the document]
Comment (bradfitz) [reply tailcfg/tailcfg.go]: whoops. copy/paste-o during earlier rewording. PTAL.

File: control/controlclient/direct.go [hunk 2]
```diff
@@ -1724,7 +1729,7 @@ func (c *Direct) ReportHealthChange(sys health.Subsystem, sysErr error) {
 	// Best effort, no logging:
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
-	res, err := np.post(ctx, "/machine/update-health", req)
+	res, err := np.post(ctx, "/machine/update-health", key.NodePublic{}, req)
```
[the remark this answers is not in the document]
Comment (bradfitz) [reply control/controlclient/direct.go]: done, and added to the request body struct.

File: tailcfg/tailcfg.go [hunk 3]
```diff
@@ -2683,3 +2683,15 @@ type EarlyNoise struct {
...
+// of it akin to an HTTP Host header or SNI. The value may be absent (notably
+// for old clients) but if present, it should match the request. An invalid
+// non-empty value will result in an error.
```
[the remark this answers is not in the document]
Comment (bradfitz) [reply tailcfg/tailcfg.go]: done

**Review by andrew-d — APPROVED** [review]

**Review by maisem — APPROVED** [review]

File: tailcfg/tailcfg.go [hunk 4]
```diff
@@ -2683,3 +2683,15 @@ type EarlyNoise struct {
...
+//
+// The possible values depend on the request path, but for /machine (Noise)
+// requests, they'll usually be a node public key, matching the Request JSON
+// body's NodeKey.
```
[the remark this answers is not in the document]
Comment (bradfitz) [reply tailcfg/tailcfg.go]: done

**Pull request merged**